### ISSUE TITLE
ignite-cli generate command leaves some unnecessary lines 

### ISSUE THREAD

**Issue by hamzakat** [infinitered/ignite#2325]
### Describe the bug

When I try to run `npx ignite-cli generate model MODEL_NAME`, the resulted file still contains some additional lines inherented from the .ejs template files like so:
![image](https://user-images.githubusercontent.com/74574779/205074132-04bfb16e-4a97-4953-815c-a58ef04ffd4d.png)

### Ignite version

8.4.2

### Additional info
```
System
  platform           win32       
  arch               x64         
  cpu                8 cores      Intel(R) Core(TM) i5-10300H CPU @ 2.50GHz
  directory          IgniteApp    C:\Users\mh\Desktop\Codes\_sketch\IgniteApp

JavaScript (and globally-installed packages)
  node                    16.14.0                                   C:\Program Files\nodejs\node.EXE
  npm                     8.3.1                                     C:\Program Files\nodejs\npm.CMD
    expo-cli              6.0.8                                    
  yarn                    1.22.15                                   C:\Program Files\nodejs\yarn.CMD                       
    create-vite           3.1.0                                    
    docsify-cli           4.4.4                                    
    expo-cli              6.0.8                                    
    react-native          0.70.2                                   
    turtle-cli            0.28.1                                   
  pnpm                    6.11.0                                    C:\Program Files\nodejs\pnpm.CMD
                          <no pnpm global package info available>  
  expo                    46.0.17                                   managed

Ignite
  ignite-cli         8.4.2        C:\Users\mh\AppData\Local\npm-cache\_npx\e31027f3785124a8\node_modules\.bin\ignite.CMD
  ignite src         build        C:\Users\mh\AppData\Local\npm-cache\_npx\e31027f3785124a8\node_modules\ignite-cli\build

Android
  java               11.0.13      C:\Program Files\Eclipse Adoptium\jdk-11.0.13.8-hotspot\bin\java.EXE
  android home       -            C:\Users\mh\AppData\Local\Android\Sdk

Tools
  git                git version 2.32.0.windows.2   C:\Program Files\Git\cmd\git.EXE 
```

**Comment by hamzakat** [comment]
I would be interested in fixing that if I get some hints on the initial bug inspection 😄

**Comment by frankcalise** [comment]
@hamzakat It's likely somewhere in `src/tools/generators.ts` would be my hunch. 

I think it's a Windows only issue from what I can tell. So, perhaps something with parsing the content out in the `frontMatter` function or maybe some path isn't proper for the Windows side.

Maybe @kateinkim or @joshuayoes might have more direction.

**Comment by kateinkim** [comment]
Seems like the front matter isn't getting parsed out on windows. Just throwing out an idea but I wonder if we should not assume that we need the `\r\n` [for all windows](https://github.com/infinitered/ignite/blob/e4ccaff0abce0a0fc6e35372b5968136164c6ed0/src/tools/generators.ts#L8)? I wonder if we should just match on any line starting with `---` regardless of line ending?

**Comment by Overdash** [comment]
Is there any work around for this?

**Comment by frankcalise** [comment]
@Overdash we have a solution for it in #2328 - likely to be released later today at some point

**Comment by Overdash** [comment]
Hmm just tired out 8.4.5 - same issue on Windows
ran `npx ignite-cli@latest generate Screen AuthScreen`

and got:
```js
---
patch:
  path: "app/screens/index.ts"
  append: "export * from \"./AuthScreen\"\n"
  skip: 
---
import React, { FC } from "react"
import { observer } from "mobx-react-lite"
....  // all the regular stuff generated
```

**Comment by frankcalise** [comment]
@Overdash can you provide the `npx ignite-cli doctor` output and provide any other info around your Windows setup? WSL, powershell etc.

Thanks!

**Comment by Overdash** [comment]
I tired it both with MINW64 and Powershell.
(running either commands with the `--debug` flag gave me the same outputs)
here are the doctor results:
```
System
  platform           win32                                                 
  arch               x64         
  cpu                8 cores      Intel(R) Core(TM) i7-4790K CPU @ 4.00GHz
  directory          TotalApp       P:\Dev\code\JS\TotalApp

JavaScript (and globally-installed packages)
  node                    16.19.0      P:\Programs\nodejs\node.EXE
  npm                     9.2.0        C:\Users\Dash\AppData\Roaming\npm\npm.CMD
    eas-cli               3.1.1       
    eslint                8.5.0       
    expo-cli              6.0.8       
    npm-windows-upgrade   6.0.1       
    npm                   9.2.0       
    yarn                  1.22.19     
  yarn                    1.22.19      C:\Users\Dash\AppData\Roaming\npm\yarn.CMD
  pnpm                    -            not installed
  expo                    47.0.10      managed

Ignite
  ignite-cli         8.4.5        C:\Users\Dash\AppData\Local\npm-cache\_npx\bec2839915ba8fc3\node_modules\.bin\ignite.CMD
  ignite src         build        C:\Users\Dash\AppData\Local\npm-cache\_npx\bec2839915ba8fc3\node_modules\ignite-cli\build

Android
  java               1.8.0_281    C:\Program Files (x86)\Common Files\Oracle\Java\javapath\java.EXE
  android home       -            undefined

Tools
  git                git version 2.36.1.windows.1   C:\Program Files\Git\cmd\git.EXE 

```

**Comment by Overdash** [comment]
@frankcalise just bumping this since the ticket is closed and might become forgotten

**Comment by joshuayoes** [comment]
Can you upload the .ejs file you have locally for us to test with?

**Comment by Overdash** [comment]
@joshuayoes I'm using the default .ejs files

```ejs
---
patch:
  path: "app/screens/index.ts"
  append: "export * from \"./<%= props.pascalCaseName %>Screen\"\n"
  skip: <%= props.skipIndexFile %>
---
import React, { FC } from "react"
import { observer } from "mobx-react-lite"
import { ViewStyle } from "react-native"
import { StackScreenProps } from "@react-navigation/stack"
import { AppStackScreenProps } from "../navigators"
import { Screen, Text } from "../components"
// import { useNavigation } from "@react-navigation/native"
// import { useStores } from "@model-utils/useStores"

// STOP! READ ME FIRST!
// To fix the TS error below, you'll need to add the following things in your navigation config:
// - Add `<%= props.pascalCaseName %>: undefined` to AppStackParamList
// - Import your screen, and add it to the stack:
//     `<Stack.Screen name="<%= props.pascalCaseName %>" component={<%= props.pascalCaseName%>Screen} />`
// Hint: Look for the 🔥!

// REMOVE ME! ⬇️ This TS ignore will not be necessary after you've added the correct navigator param type
// @ts-ignore
export const <%= props.pascalCaseName %>Screen: FC<StackScreenProps<AppStackScreenProps, "<%= props.pascalCaseName %>">> = observer(function <%= props.pascalCaseName %>Screen() {
  // Pull in one of our MST stores
  // const { someStore, anotherStore } = useStores()

  // Pull in navigation via hook
  // const navigation = useNavigation()
  return (
    <Screen style={$root} preset="scroll">
      <Text text="<%= props.camelCaseName %>" />
    </Screen>
  )
})

const $root: ViewStyle = {
  flex: 1,
}

```

**Comment by Overdash** [comment]
@joshuayoes any updates on this?

**Comment by joshuayoes** [comment]
@Overdash A few of us at IR have tried to recreate this on our windows machines, but unfortunately haven't been able to recreate it, which makes it challenging for us to fix your issue. 

These generators work by [reading the EJS files, parsing the front matter, and then doing a find and replace with props provided from the frontmatter](https://github.com/infinitered/ignite/blob/master/src/tools/generators.ts#L224-L290).

The issue here seems to be that the new line character is not getting properly parsed on Windows. Originally, we were using only Unix new lines characters to split the lines, but not windows. However, that should be addressed in [this PR by using the End of Line constant](https://github.com/infinitered/ignite/pull/2328).

Perhaps you could try to update the new line character and run the CLI locally from the repo to see if that is the cause?

**Comment by tank104** [comment]
Just started using Ignite (and ReactNative) today - so clean install of the latest, and I am getting this on windows too.

**Comment by ben186** [comment]
@joshuayoes @tank104 It works if the eol format for the template files (*.ejs) change from LF to CRLF on windows. The generator probably skips those beginning lines due to different new line character.

**Comment by smitroshin** [comment]
I fixed the issue on my Windows machine via next workaround:

1. `git config --global core.autocrlf input` or `git config core.autocrlf input`
2. Override line endings of entire `ejs` files from LF to CRLF.
That can be done in different ways. I used VS Code.
![image](https://github.com/infinitered/ignite/assets/20210690/f73a25a1-cfeb-4d05-88a4-25a6dfd06d77)

P.S. I don't know if that won't happen again in the future, like after next `git pull` (when someone of my mates that uses Linux will do new changes in those files or will create new generator templates.). So, I'm unsure if it's a permanent fix.
